File: src/dataset/python/robot-simulator_exercism-ThomasZumsteg/gemini/GeminiFlash_robot_simulator_v1.py
# NORTH, EAST, SOUTH, WEST are directions a robot can face
NORTH, EAST, SOUTH, WEST = 0, 1, 2, 3

class Robot:
    """Robot is a simulation of a robot"""

    # Commands that the robot can be given
    _valid_commands = {
        'R': 'turn_right',
        'L': 'turn_left',
        'A': 'advance',
    }

    _movements = {
        NORTH: (0, 1),
        SOUTH: (0, -1),
        EAST: (1, 0),
        WEST: (-1, 0),
    }

    def __init__(self, bearing=NORTH, x=0, y=0):
        """__init__ creates the robot, defaults to facing north at the origin"""
        self.bearing = bearing
        self.coordinates = (x, y)

    def turn_right(self):
        """turn_right turns the robot to the right"""
        self.bearing = (self.bearing + 1) % 4

    def turn_left(self):
        """turn_left turns the robot to the left"""
        self.bearing = (self.bearing - 1) % 4

    def advance(self):
        """advance moves the robot one space forward in the direction it's facing"""
        dx, dy = Robot._movements[self.bearing]
        x, y = self.coordinates
        self.coordinates = (x + dx, y + dy)

    def simulate(self, commands):
        """simulate give the robot a set of commands"""
        command_map = {
            'R': self.turn_right,
            'L': self.turn_left,
            'A': self.advance,
        }
        for c in commands:
            command_map[c]()

File: src/dataset/javascript/robot-simulator_exercism-ThomasZumsteg/gemini/GeminiFlash_robot-simulator_v1.js
/* A robot to that follows simple instructions */
var Robot = function() {
    this.bearing = null;
    this.coordinates = null;
}

// Directions the robot can face and associated vectors
var directions = {
    'north': 0,
    'east': 1,
    'south': 2,
    'west': 3
};
var vectors = [[0, 1], [1, 0], [0, -1], [-1, 0]];

Robot.prototype.orient = function(direction) {
    /* Sets the direction of the robot */
    if (!(direction in directions)) {
        throw "Invalid Robot Bearing";
    }
    this.bearing = direction;
};

Robot.prototype.turnRight = function() {
    /* Changes the bearing one step to the right */
    const currentDirectionIndex = directions[this.bearing];
    const newDirectionIndex = (currentDirectionIndex + 1) % 4;
    this.bearing = Object.keys(directions).find(key => directions[key] === newDirectionIndex);
};

Robot.prototype.turnLeft = function() {
    /* Changes the bearing one step to the left */
    const currentDirectionIndex = directions[this.bearing];
    const newDirectionIndex = (currentDirectionIndex + 3) % 4;
    this.bearing = Object.keys(directions).find(key => directions[key] === newDirectionIndex);
};

/* Sets the robots position */
Robot.prototype.at = function(x, y) { this.coordinates = [x,y] };

Robot.prototype.advance = function() {
    /* Move the position one step in the direction the robot is facing */
    const dirIndex = directions[this.bearing];
    const vector = vectors[dirIndex];
    this.coordinates[0] += vector[0];
    this.coordinates[1] += vector[1];
};

// Instructions the robot understands
var instructions = {
    'L': 'turnLeft',
    'R': 'turnRight',
    'A': 'advance',
};

Robot.prototype.instructions = function(instruction_list) {
    /* Make a list of instructions from a string */
    const instructionsArray = [];
    for (let i = 0; i < instruction_list.length; i++) {
        const instruction = instruction_list[i];
        instructionsArray.push(instructions[instruction]);
    }
    return instructionsArray;
};

Robot.prototype.place = function(args) {
    /* Initilized the direction and location of a robot */
    this.orient(args['direction']);
    this.at(args['x'], args['y']);
};

Robot.prototype.evaluate = function(instruction_list) {
    /* Robot preforms a series of instructions */
    const actions = this.instructions(instruction_list);
    for (let i = 0; i < actions.length; i++) {
        this[actions[i]]();
    }
};

module.exports = Robot;
exports.directions = Object.keys(directions);